MSAL/src/instance/MSALDeviceInfoProvider.m

```diff
@@ -32,59 +32,85 @@
 #import "MSIDRequestParameters+Broker.h"
 #import "MSALDeviceInformation+Internal.h"
 
+#import "MSIDWorkPlaceJoinConstants.h"
+#import "MSIDWorkPlaceJoinUtil.h"
+#import "MSIDRegistrationInformation.h"
+
 @implementation MSALDeviceInfoProvider
 
 - (void)deviceInfoWithRequestParameters:(MSIDRequestParameters *)requestParameters
-                        completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock API_AVAILABLE(ios(13.0), macos(10.15))
+                        completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock
 {
     if (![requestParameters shouldUseBroker])
     {
         NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorBrokerNotAvailable, @"Broker is not enabled for this operation. Please make sure you have enabled broker support for your application", nil, nil, nil, nil, nil, YES);
         completionBlock(nil, error);
         return;
     }
-    
-    
-    if (![MSIDSSOExtensionGetDeviceInfoRequest canPerformRequest])
-    {
-        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, requestParameters, @"Broker is not present on this device. Defaulting to personal mode");
-        
-        MSALDeviceInformation *msalDeviceInfo = [MSALDeviceInformation new];
-        msalDeviceInfo.deviceMode = MSALDeviceModeDefault;
-        completionBlock(msalDeviceInfo, nil);
-        return;
-    }
-    
-    NSError *requestError;
-    MSIDSSOExtensionGetDeviceInfoRequest *ssoExtensionRequest = [[MSIDSSOExtensionGetDeviceInfoRequest alloc] initWithRequestParameters:requestParameters
-                                                                                                                                  error:&requestError];
-    
-    if (!ssoExtensionRequest)
-    {
-        completionBlock(nil, requestError);
-        return;
-    }
-    
-    if (![self setCurrentSSOExtensionRequest:ssoExtensionRequest])
-    {
-        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Trying to start a get accounts request while one is already executing", nil, nil, nil, nil, nil, YES);
-        completionBlock(nil, error);
-        return;
-    }
-    
-    [ssoExtensionRequest executeRequestWithCompletion:^(MSIDDeviceInfo * _Nullable deviceInfo, NSError * _Nullable error)
+
+    if (@available(iOS 13.0, macOS 10.15, *))
     {
-        [self copyAndClearCurrentSSOExtensionRequest];
-        
-        if (!deviceInfo)
+        if (![MSIDSSOExtensionGetDeviceInfoRequest canPerformRequest])
```

Comment:
I think you can simplify some of the logic by combining iOS 13 and this check.

e.g. BOOL canCallSSOExtension = NO;

if (iOS 13, macOS 10.15) {
         canCallSSOextension = [MSIDSSOExtensionGetDeviceInfoRequest canPerformRequest]
}

if (!canCallSSOExtension) {

// create default info
// call completionBlock
}

Reply:
Thanks for the suggestion, refactor function accordingly.